### PR TITLE
etcd: Update version to 3.5.9

### PR DESCRIPTION
**Pull request by humblec**
- [x] update etcd version to 3.5.9
- [x] use etcd v3.5.9 in the installation script
- [x]  use upstream etcd github path instead of redirecting one
    
    In the installation script we use coreos/etcd path which redirect
    to etcd-io/etcd. This commit replace the same.

/kind cleanup

Depends on #https://github.com/kubernetes/kubernetes/pull/117999

### CODE REVIEW

**Comment by pacoxu** [comment]
There are several places to be updated for etcd version. You can follow https://github.com/kubernetes/kubernetes/pull/117335.

**Comment by kkkkun** [comment]
> @kkkkun can you add this PR too in the list?

Done! Thanks

**Comment by kkkkun** [comment]
In my options. You should merge this change to ` update etcd version to 3.5.9` just like https://github.com/kubernetes/kubernetes/pull/117335

**Comment by kkkkun** [comment]
> > #117335
> 
> You mean update this PR with the other changes required for this bump @kkkkun ?

No. It‘s ok for this.

**Comment by pacoxu** [comment]
There is a check on versions:

https://github.com/kubernetes/kubernetes/blob/8d491e676ab9e4c84d68e28d727781bd5cc69f89/build/dependencies.yaml#L63-L80

You need to align them all to make the CI green.

**Comment by kkkkun** [comment]
Title should be `etcd: Update version to 3.5.9`. Just like https://github.com/kubernetes/kubernetes/pull/117335

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -482,6 +482,7 @@ var (
 		25: "3.5.8-0",
 		26: "3.5.8-0",
 		27: "3.5.8-0",
```
Comment (kkkkun) [inline cmd/kubeadm/app/constants/constants.go:484]: Should we replace `3.5.8-0` to`3.5.9-0` from 22-27?

Comment (kkkkun) [reply cmd/kubeadm/app/constants/constants.go:484]: @dims  @jpbetz  @pacoxu  P;ease look this change.

**Review by pacoxu — APPROVED** [review]
/lgtm

**Pull request merged**